`cargo install mdbook-epub` is broken

When I run `cargo install mdbook-epub` I get this compile error:
```
   Compiling mdbook-epub v0.4.31
error[E0277]: `?` couldn't convert the error to `Error`
  --> /Users/njn/.cargo/registry/src/index.crates.io-6f17d22bba15001f/mdbook-epub-0.4.31/src/config.rs:34:63
   |
34 |                 let mut cfg: Config = table.clone().try_into()?;
   |                                                               ^ the trait `From<toml::de::Error>` is not implemented for `Error`
   |
   = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
   = help: the following other types implement trait `From<T>`:
             <Error as From<SemVerError>>
             <Error as From<ReqParseError>>
             <Error as From<Box<ureq::Error>>>
             <Error as From<epub_builder::Error>>
             <Error as From<mdbook::errors::Error>>
             <Error as From<ureq::Error>>
             <Error as From<RenderError>>
             <Error as From<toml::de::Error>>
             <Error as From<std::io::Error>>
   = note: required for `Result<config::Config, Error>` to implement `FromResidual<Result<Infallible, toml::de::Error>>`
```
I see this with rust 1.71 and nightly 1.73.

This caused CI breakage for me here: https://github.com/nnethercote/perf-book/actions/runs/5572424446/jobs/10178452751. I will have to disable the epub build for the Rust perf-book because of this :(

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -2,7 +2,7 @@
 authors = ["Michael Bryan <michaelfbryan@gmail.com>"]
 description = "An EPUB renderer for mdbook."
 name = "mdbook-epub"
-version = "0.4.31"
+version = "0.4.32-beta1"
 readme = "README.md"
 license = "MIT"
 repository = "https://github.com/Michael-F-Bryan/mdbook-epub"
@@ -24,20 +24,20 @@ doc = false
 
 [dependencies]
 epub-builder = "0.5"
-thiserror = "1.0.40"
+thiserror = "1.0.43"
 pulldown-cmark = "0.9.3"
-semver = "0.11"
-serde = "1.0.17"
+semver = "1.0.17"
+serde = { version = "1.0.163", features = ["derive"] }
 serde_derive = "1.0"
 serde_json = "1.0.96"
 structopt = "0.3.26"
 mime_guess = "2.0"
 env_logger = "0.10.0"
 log = "0.4.17"
-mdbook = { version = "0.4.31", default-features = false }
+mdbook = { version = "0.4.32", default-features = false }
 handlebars = "4.3.7"
-toml = "0.5.11"
-html_parser = "0.6.2"
+toml = "0.7.6"
+html_parser = "0.7.0"
 url = "2.3"
 ureq = "2.5"
 const_format = "0.2.30"
@@ -45,6 +45,6 @@ const_format = "0.2.30"
 [dev-dependencies]
 tempdir = "0.3.7"
 epub = "2.1.1"
-serial_test = "0.5"
+serial_test = "2.0.0"
 mockall = "0.11"
 env_logger = "*"

--- a/src/bin/mdbook-epub.rs
+++ b/src/bin/mdbook-epub.rs
@@ -65,8 +65,8 @@ struct Args {
     #[structopt(
         short = "s",
         long = "standalone",
-        parse(try_from_str),
-        default_value = "false",
+        // parse(try_from_str), // probably causes mdbook-epub install error
+        // default_value = "false",
         help = "Run standalone (i.e. not as a mdbook plugin)"
     )]
     standalone: bool,

--- a/src/bin/mdbook-epub.rs
+++ b/src/bin/mdbook-epub.rs
@@ -65,8 +65,8 @@ struct Args {
     #[structopt(
         short = "s",
         long = "standalone",
-        // parse(try_from_str), // probably causes mdbook-epub install error
-        // default_value = "false",
+        parse(try_from_str), // can that cause mdbook-epub install error??
+        default_value = "false", // can that cause mdbook-epub install error??
         help = "Run standalone (i.e. not as a mdbook plugin)"
     )]
     standalone: bool,

--- a/src/config.rs
+++ b/src/config.rs
@@ -71,3 +71,33 @@ impl Default for Config {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::path::Path;
+    use super::*;
+
+    #[test]
+    fn test_from_render_context_minimal_settings() {
+        let json = ctx_with_template(
+            "unknown_src",
+            tempdir::TempDir::new("test-mdbook-epub").unwrap().path(),
+        ).to_string();
+        let ctx = RenderContext::from_json(json.as_bytes()).unwrap();
+        let config = Config::from_render_context(&ctx);
+        assert!(config.is_ok());
+    }
+
+    fn ctx_with_template(source: &str, destination: &Path) -> serde_json::Value {
+        json!({
+            "version": mdbook::MDBOOK_VERSION,
+            "root": "tests/dummy",
+            "book": {"sections": [], "__non_exhaustive": null},
+            "config": {
+                "book": {"authors": [], "language": "en", "multilingual": false,
+                    "src": source, "title": "DummyBook"},
+                "output": {"epub": {"optional": true}}},
+            "destination": destination
+        })
+    }
+}

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -91,6 +91,11 @@ impl From<ureq::Error> for Error {
         Error::HttpError(Box::new(e))
     }
 }
+impl From<Box<ureq::Error>> for Error {
+    fn from(e: Box<ureq::Error>) -> Self {
+        Error::HttpError(e)
+    }
+}
 
 /// The exact version of `mdbook` this crate is compiled against.
 pub const MDBOOK_VERSION: &str = mdbook::MDBOOK_VERSION;

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -91,11 +91,6 @@ impl From<ureq::Error> for Error {
         Error::HttpError(Box::new(e))
     }
 }
-impl From<Box<ureq::Error>> for Error {
-    fn from(e: Box<ureq::Error>) -> Self {
-        Error::HttpError(e)
-    }
-}
 
 /// The exact version of `mdbook` this crate is compiled against.
 pub const MDBOOK_VERSION: &str = mdbook::MDBOOK_VERSION;

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -73,9 +73,7 @@ pub enum Error {
     #[error(transparent)]
     Book(#[from] mdbook::errors::Error),
     #[error(transparent)]
-    Semver(#[from] semver::SemVerError),
-    #[error(transparent)]
-    SemverReqParse(#[from] semver::ReqParseError),
+    Semver(#[from] semver::Error),
     #[error(transparent)]
     EpubBuilder(#[from] epub_builder::Error),
     #[error(transparent)]

